.github/workflows/k8s-patch-os-jobs.yml

```
@@ -0,0 +1,173 @@
+        run: |
+          SECRET_NAMES=(
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-master-node-endpoint-pending-value"
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-ssh-key-pending-value"
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-master-node-endpoint"
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-ssh-key"
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-master-node-endpoint-temporary-storage"
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-ssh-key-temporary-storage"
+          )
+
+          for SECRET_NAME in "${SECRET_NAMES[@]}"; do
+            echo "Checking secret: $SECRET_NAME"
+
+
+            if aws secretsmanager describe-secret --secret-id $SECRET_NAME; then
+              echo "Success: The secret exists."
+            else
+              echo "Secret does not exist. Creating new secret"
+              aws secretsmanager create-secret --name $SECRET_NAME --secret-string "empty"
+            fi
+          done
+
+
+      - name: Save Private Key and Master Public Endpoint to Secret Manager
+        working-directory: .github/workflows/util
+        run: |
+          SECRET_NAMES=(
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-master-node-endpoint-pending-value"
+            "e2e-test/${{ env.REPO }}/${{ env.LANGUAGE }}-k8s-ssh-key-pending-value"
+          )
+
+          private_key_file=k8s-on-ec2-${{ env.EC2_NAME }}-${{ env.LANGUAGE }}-key-pair-${{ env.TESTING_ID }}.pem
+          private_key_content=$(<"$private_key_file")
+
+          master_public_ip=$(aws ec2 describe-instances \
+            --filters "Name=tag:Name,Values=k8s-on-ec2-${{ env.EC2_NAME }}-${{ env.LANGUAGE }}-master-${{ env.TESTING_ID }}" "Name=instance-state-name,Values=running" \
+            --query "Reservations[*].Instances[*].PublicIpAddress" \
+            --output text)
+
+          aws secretsmanager update-secret --secret-id "${SECRET_NAMES[0]}" --secret-string $master_public_ip
+          aws secretsmanager update-secret --secret-id "${SECRET_NAMES[1]}" --secret-string "$private_key_content"
+  
+  run-java-k8s-test:
+    if: ${{ inputs.LANGUAGE == 'java' || inputs.LANGUAGE == 'all' }}
+    needs: create-k8s-on-ec2
+    uses: ./.github/workflows/java-k8s-e2e-test.yml
+    secrets: inherit
+    with:
+      aws-region: 'us-east-1'
+      caller-workflow-name: 'k8s-os-patching'
+      caller-repository: ${{ inputs.repo }}
+
+  run-python-k8s-test:
+    if: ${{ inputs.LANGUAGE == 'python' }}
+    needs: create-k8s-on-ec2
+    uses: ./.github/workflows/python-k8s-e2e-test.yml
+    secrets: inherit
+    with:
+      aws-region: 'us-east-1'
+      caller-workflow-name: 'k8s-os-patching'
+      caller-repository: ${{ inputs.repo }}
+
+  update-secrets:
+    needs: [ run-java-k8s-test, run-python-k8s-test ]
+    if: ${{ always() && (needs.run-java-k8s-test.result == 'success' || needs.run-python-k8s-test.result == 'success') }}
```

Shouldn't this be an AND operation? Or done separately for each language?
```suggestion
    if: ${{ always() && (needs.run-java-k8s-test.result == 'success' && needs.run-python-k8s-test.result == 'success') }}
```

---

It needs to be an `||` operation because only k8s-java or k8s-python test will run depending on which language the instance is for.

---

Understood, ty